From 0000000000000000000000000000000000000000 Fri Jul 18 00:00:00 2025
From: Nathanael Landais <nathanael@lenewt.com>
Date: Fri, 18 Jul 2025 15:32:54 +0200
Subject: [PATCH] Move unsigned integer operators inline in QDataStream class

Move the unsigned integer operator implementations from separate inline 
functions at the bottom of the file to be directly inline in the class 
declaration. This improves code organization and potentially compilation 
performance.

diff --git a/src/corelib/io/qdatastream.h b/src/corelib/io/qdatastream.h
index 1234567..8901234 100644
--- a/src/corelib/io/qdatastream.h
+++ b/src/corelib/io/qdatastream.h
@@ -145,13 +145,9 @@
     void setVersion(int);
 
     QDataStream &operator>>(qint8 &i);
-    QDataStream &operator>>(quint8 &i);
     QDataStream &operator>>(qint16 &i);
-    QDataStream &operator>>(quint16 &i);
     QDataStream &operator>>(qint32 &i);
-    QDataStream &operator>>(quint32 &i);
     QDataStream &operator>>(qint64 &i);
-    QDataStream &operator>>(quint64 &i);
     QDataStream &operator>>(std::nullptr_t &ptr) { ptr = nullptr; return *this; }
 
     QDataStream &operator>>(bool &i);
@@ -161,13 +157,9 @@
     QDataStream &operator>>(char *&str);
 
     QDataStream &operator<<(qint8 i);
-    QDataStream &operator<<(quint8 i);
     QDataStream &operator<<(qint16 i);
-    QDataStream &operator<<(quint16 i);
     QDataStream &operator<<(qint32 i);
-    QDataStream &operator<<(quint32 i);
     QDataStream &operator<<(qint64 i);
-    QDataStream &operator<<(quint64 i);
     QDataStream &operator<<(std::nullptr_t) { return *this; }
     QDataStream &operator<<(bool i);
     QDataStream &operator<<(qfloat16 f);
@@ -175,6 +167,17 @@
     QDataStream &operator<<(double f);
     QDataStream &operator<<(const char *str);
 
+    // Inline implementations for unsigned integer operators
+    QDataStream &operator>>(quint8 &i) { return *this >> reinterpret_cast<qint8&>(i); }
+    QDataStream &operator>>(quint16 &i) { return *this >> reinterpret_cast<qint16&>(i); }
+    QDataStream &operator>>(quint32 &i) { return *this >> reinterpret_cast<qint32&>(i); }
+    QDataStream &operator>>(quint64 &i) { return *this >> reinterpret_cast<qint64&>(i); }
+    
+    QDataStream &operator<<(quint8 i) { return *this << qint8(i); }
+    QDataStream &operator<<(quint16 i) { return *this << qint16(i); }
+    QDataStream &operator<<(quint32 i) { return *this << qint32(i); }
+    QDataStream &operator<<(quint64 i) { return *this << qint64(i); }
+
     QDataStream &readBytes(char *&, uint &len);
     int readRawData(char *, int len);
 
@@ -337,30 +340,6 @@
 inline void QDataStream::setVersion(int v)
 { ver = v; }
 
-inline QDataStream &QDataStream::operator>>(quint8 &i)
-{ return *this >> reinterpret_cast<qint8&>(i); }
-
-inline QDataStream &QDataStream::operator>>(quint16 &i)
-{ return *this >> reinterpret_cast<qint16&>(i); }
-
-inline QDataStream &QDataStream::operator>>(quint32 &i)
-{ return *this >> reinterpret_cast<qint32&>(i); }
-
-inline QDataStream &QDataStream::operator>>(quint64 &i)
-{ return *this >> reinterpret_cast<qint64&>(i); }
-
-inline QDataStream &QDataStream::operator<<(quint8 i)
-{ return *this << qint8(i); }
-
-inline QDataStream &QDataStream::operator<<(quint16 i)
-{ return *this << qint16(i); }
-
-inline QDataStream &QDataStream::operator<<(quint32 i)
-{ return *this << qint32(i); }
-
-inline QDataStream &QDataStream::operator<<(quint64 i)
-{ return *this << qint64(i); }
-
 template <typename Enum>
 inline QDataStream &operator<<(QDataStream &s, QFlags<Enum> e)
 { return s << e.i; }
